src/test/java/htsjdk/samtools/BamFileIoUtilsUnitTest.java

```diff
@@ -0,0 +1,103 @@
+package htsjdk.samtools;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.beta.exception.HtsjdkException;
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
+public class BamFileIoUtilsUnitTest extends HtsjdkTest {
+    final static String TEST_DATA_DIR = "src/test/resources/htsjdk/samtools/cram/";
+    final static String NA12878_8000 = TEST_DATA_DIR + "CEUTrio.HiSeq.WGS.b37.NA12878.20.first.8000.bam";
+    final static String NA12878_20_21 = TEST_DATA_DIR + "NA12878.20.21.unmapped.orig.bam";
+    final static int DEFAULT_NUM_RECORDS_TO_COMPARE = 10;
+
+    @Test
+    public void testReheaderBamFile(){
+        final File originalBam = new File(this.NA12878_8000);
+        SAMFileHeader header = SamReaderFactory.make().getFileHeader(new File(NA12878_20_21));
+        try {
+            final File output = File.createTempFile("output", ".bam");
+            BamFileIoUtils.reheaderBamFile(header, originalBam.toPath(), output.toPath());
+
+            // Confirm that the header has been replaced
+            final SamReader outputReader = SamReaderFactory.make().open(output.toPath());
+            Assert.assertEquals(outputReader.getFileHeader(), header);
+
+            // Confirm that the reads are the same as the original
+            Assert.assertTrue(compareBamReads(originalBam, output, DEFAULT_NUM_RECORDS_TO_COMPARE));
+        } catch (IOException e){
+            throw new HtsjdkException("Could not create a temporary output file.", e);
```

Comment:
You can just add throws IOException to the test method.  It's not clear that this error message will be correct, there are a bunch of IO operations going on during this method so it could be more confusing than helpful.

Reply:
done